### PR TITLE
Document CUTENSOR_PATH

### PR DESCRIPTION
**Pull request by kmaehashi**
Document new build-time environment variable introduced in #2210

### CODE REVIEW

**Comment by toslunar** [comment]
Jenkins, test this please.

File: docs/source/reference/environment.rst [hunk 1]
```diff
@@ -46,10 +46,13 @@ Here are the environment variables CuPy uses.
...
-| ``CUDA_PATH`` | See the description above.                                          |
-+---------------+---------------------------------------------------------------------+
-| ``NVCC``      | Define the compiler to use when compiling CUDA files.               |
-+---------------+---------------------------------------------------------------------+
+These environment variables are used during installation (building CuPy from source).
+
++-------------------+---------------------------------------------------------------------+
+| ``CUDA_PATH``     | See the description above.                                          |
++-------------------+---------------------------------------------------------------------+
+| ``CUTENSOR_PATH`` | Path to the cuTENSOR v0.2+ root containing ``lib`` and ``include``  |
```
Comment (toslunar) [inline docs/source/reference/environment.rst]: Is the minimum version relevant to paths to `lib` and `include`?

Comment (kmaehashi) [reply docs/source/reference/environment.rst]: No, it's just to indicate the minimum required version in https://github.com/cupy/cupy/pull/2341.

Comment (toslunar) [reply docs/source/reference/environment.rst]: How about adding the info to `docs/source/install.rst#optional-libraries`?

**Comment by kmaehashi** [comment]
Thanks, fixed as suggested.

**Comment by toslunar** [comment]
Jenkins, test this please.

**Comment by toslunar** [comment]
The Jenkins failure is unrelated to the PR.

**Pull request merged**